tests/api/test_distribution.py

```diff
@@ -58,3 +59,39 @@ def test_post_all_distributions_failure(self, query_mock):
         url = url_for('api.distributions.distributions', date="2015-10-01")
         response = self.client.post(url)
         assert_equal(response.status_code, 400)
+
+    def test_multiplex_directory_tiles(self):
+        tile1 = {"type": "sponsored", "directoryId": 1, "url": "http://www.test.com"}
+        tile2 = {"type": "sponsored", "directoryId": 2, "url": "http://www.test1.com"}
+        tile3 = {"type": "affiliate", "directoryId": 3, "url": "http://www.mozilla.org"}
+        tile4 = {"type": "affiliate", "directoryId": 4, "url": "http://www.mozilla.org"}
+        tile5 = {"type": "affiliate", "directoryId": 5, "url": "http://developer.mozilla.org"}
+        tiles = [tile1, tile2, tile3, tile4, tile5]
+        ret = list(multiplex_directory_tiles(tiles))
+        assert_equal(len(ret), 4)
+        assert_equal([
+            [tile3, tile5, tile1],
+            [tile4, tile5, tile1],
+            [tile3, tile5, tile2],
+            [tile4, tile5, tile2]
```

Comment:
heck, that was TLD+2? My apologies for the confusion. So I kept using this term wrongly. Anyway, that's how Firefox groups the url histories.